### PR TITLE
Unify OTLP path parsing/default Logic

### LINKED ISSUE #2634: Unify OTLP path parsing/default logic
_Originally posted by @dmathieu in https://github.com/open-telemetry/opentelemetry-go/pull/2625#discussion_r812657958_

### PR DESCRIPTION
**Pull request by hanyuancheung**
Fix #2634 and https://github.com/open-telemetry/opentelemetry-go/pull/2625#discussion_r813124976
- Unify of path parsing/default functionally.

### CODE REVIEW

File: exporters/otlp/internal/config.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"strings"
+)
+
+// CleanPath returns cleaned URL path. Replace with default path if path is nil
+func CleanPath(URLPath string, defaultPath string) string {
+	tmp := strings.TrimSpace(URLPath)
+	if tmp == "" {
+		return defaultPath
+	}
+	tmp = path.Clean(tmp)
```
Comment (MrAlias) [inline exporters/otlp/internal/config.go:32]: The clean operation can result in an empty string (i.e. `/dir/..`) which is returned as `"."`.

```suggestion
	tmp := path.Clean(strings.TrimSpace(URLPath))
	if tmp == "." {
		return defaultPath
	}
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - For tracestate's members, prepend the new element and remove the oldest one, which is over capacity (#2592)
 - Add event and link drop counts to the exported data from the `oltptrace` exporter. (#2601)
+- Unify of path parsing functionally in `otlpmetric` and `otlptrace` config. (#2639)
```
Comment (MrAlias) [inline CHANGELOG.md:26]:
```suggestion
- Unify path cleaning functionally in the `otlpmetric` and `otlptrace` config. (#2639)
```

File: exporters/otlp/internal/config_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		URLPath     string
+		defaultPath string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "test-clean-nil-path",
```
Comment (MrAlias) [inline exporters/otlp/internal/config_test.go:30]:
```suggestion
			name: "test-clean-empty-path",
```

File: exporters/otlp/internal/config_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			want: "/prefix/v1/metrics",
+		},
+		{
+			name: "test-clean-traces-path",
+			args: args{
+				URLPath:     "https://env_endpoint",
+				defaultPath: "DefaultTracesPath",
+			},
+			want: "/https:/env_endpoint",
+		},
```
Comment (MrAlias) [inline exporters/otlp/internal/config_test.go:52]:
```suggestion
		},
		{
			name: "spaces trimmed",
			args: args{
				URLPath:     " /dir",
			},
			want: "/dir",
		},
		{
			name: "clean path empty",
			args: args{
				URLPath:     "dir/..",
				defaultPath: "DefaultTracesPath",
			},
			want: "DefaultTracesPath",
		},
		{
			name: "make absolute",
			args: args{
				URLPath:     "dir/a",
			},
			want: "/dir/a",
		},
```

**Review by MrAlias — APPROVED** [review]

**Comment by hanyuancheung** [comment]
PTLA.

File: exporters/otlp/internal/config_test.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		URLPath     string
+		defaultPath string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "test-clean-empty-path",
```
Comment (XSAM) [inline exporters/otlp/internal/config_test.go:30]:
```suggestion
			name: "clean empty path",
```

File: exporters/otlp/internal/config_test.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import "testing"
+
+func TestCleanPath(t *testing.T) {
+	type args struct {
+		URLPath     string
```
Comment (XSAM) [inline exporters/otlp/internal/config_test.go:21]:
```suggestion
		urlPath     string
```

File: exporters/otlp/internal/config_test.go [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+		{
+			name: "test-clean-empty-path",
+			args: args{
+				URLPath:     "",
+				defaultPath: "DefaultPath",
+			},
+			want: "DefaultPath",
+		},
+		{
+			name: "test-clean-metrics-path",
```
Comment (XSAM) [inline exporters/otlp/internal/config_test.go:38]:
```suggestion
			name: "clean metrics path",
```

File: exporters/otlp/internal/config_test.go [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+			args: args{
+				URLPath:     "/prefix/v1/metrics",
+				defaultPath: "DefaultMetricsPath",
+			},
+			want: "/prefix/v1/metrics",
+		},
+		{
+			name: "test-clean-traces-path",
```
Comment (XSAM) [inline exporters/otlp/internal/config_test.go:46]:
```suggestion
			name: "clean traces path",
```

File: exporters/otlp/internal/config.go [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+// Package internal contains common functionality for all OTLP exporters.
+package internal // import "go.opentelemetry.io/otel/exporters/otlp/internal"
+
+import (
+	"fmt"
+	"path"
+	"strings"
+)
+
+// CleanPath returns URLPath with all spaces trimmed and all redundancies removed. If URLPath is empty or cleaning it results in an empty string, defaultPath is returned instead.
```
Comment (XSAM) [inline exporters/otlp/internal/config.go:24]:
```suggestion
// CleanPath returns a path with all spaces trimmed and all redundancies removed. If urlPath is empty or cleaning it results in an empty string, defaultPath is returned instead.
```

File: exporters/otlp/internal/config.go [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+func CleanPath(URLPath string, defaultPath string) string {
+	tmp := path.Clean(strings.TrimSpace(URLPath))
```
Comment (XSAM) [inline exporters/otlp/internal/config.go:26]:
```suggestion
func CleanPath(urlPath string, defaultPath string) string {
	tmp := path.Clean(strings.TrimSpace(urlPath))
```

**Review by XSAM — APPROVED** [review]

**Pull request merged**